### PR TITLE
Add docs for AsyncInterfaces

### LINKED ISSUE #87722: Microsoft.Bcl.AsyncInterfaces is missing API documentation
See https://github.com/dotnet/runtime/pull/84917 for more context.

The compiler generated XML file for Microsoft.Bcl.AsyncInterfaces is shipping to customers via the nuget package and is missing public API documentation. The above linked PR disables the compiler error CS1591 until all the API is documented.

When working on this, remove the NoWarn=1591 setting from the project file.

### PR DESCRIPTION
**Pull request by ericstj**
Fixes #87722 
I just copied these from the shipping XML docs.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConfiguredAsyncDisposable.cs [hunk 1]
```diff
@@ -18,6 +18,8 @@ internal ConfiguredAsyncDisposable(IAsyncDisposable source, bool continueOnCaptu
             _continueOnCapturedContext = continueOnCapturedContext;
         }
 
+        /// <summary>Asynchronously releases the unmanaged resources used by the <see cref="T:System.Runtime.CompilerServices.ConfiguredAsyncDisposable" />.</summary>
+        /// <returns>A task that represents the asynchronous dispose operation.</returns>
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConfiguredAsyncDisposable.cs:22]: It's not actually a task.  But I see this matches what we have documented, so, fine for now.

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConfiguredAsyncDisposable.cs (hunk 1 above)
Comment (carlossanlop) [reply src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConfiguredAsyncDisposable.cs:22]: I've seen other places where we say "A task that represents" when the returned value is the result of a ConfigureAwait. 

For future reference, what description would you prefer to use in plain English for these cases? Would this work?:

"An awaitable, configured value task that represents..."

Or would you use something else?

**Review by carlossanlop — APPROVED** [review]
Thanks, @ericstj!

**Comment by ericstj** [comment]
Failure is known issue https://github.com/dotnet/runtime/issues/35066
